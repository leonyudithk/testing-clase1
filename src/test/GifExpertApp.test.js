import { shallow } from 'enzyme';
import React from 'react';
import { GifExpertApp } from '../GifExpertApp';

describe('Prueba en Componente GifExpertApp', () => {
        test('mostrarme categoria correctamente', () => {
            //preparacion area de trabajo
               const wrapper = shallow(<GifExpertApp/>)
           
               // actuar en la prueba
            expect(wrapper).toMatchSnapshot()
        })
  
        //---------Prueba de categorias---//
        test('debe mostrar una lista de categorias', () => {
            //prepar la mesa de trabajo -  Arregla
             const categories = ['Dragon Ball', 'Ponys', 'animales', 'plantas', '123ssss']
             const wrapper = shallow(<GifExpertApp defaultCategories={categories}/>)
         
             // Actuar - hacer la prueba
             //realice la captura foto 
             expect(wrapper).toMatchSnapshot()

             //voy a probar que las categorias de la busqueda si existen dentro del elemento
            expect(wrapper.find('GifGrid').length).toBe(categories.length)

      })

})