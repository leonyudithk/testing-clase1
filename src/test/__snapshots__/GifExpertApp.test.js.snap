// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en Componente GifExpertApp debe mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
    <GifGrid
      category="Ponys"
      key="Ponys"
    />
    <GifGrid
      category="animales"
      key="animales"
    />
    <GifGrid
      category="plantas"
      key="plantas"
    />
    <GifGrid
      category="123ssss"
      key="123ssss"
    />
  </ol>
</Fragment>
`;

exports[`Prueba en Componente GifExpertApp mostrarme categoria correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
